attribute-picker.js

```diff
@@ -418,9 +418,24 @@ class AttributePicker extends RtlMixin(Localizer(LitElement)) {
 				if (this._dropdownIndex >= 0 && this._dropdownIndex < list.length) {
 					this.addAttribute(list[this._dropdownIndex].text);
 				} else if (this.allowFreeform) {
-					const trimmedAttribute =  this._text.trim();
-					if (trimmedAttribute.length > 0 && !this.attributeList.includes(trimmedAttribute)) {
-						this.addAttribute(this._text.trim());
+					let attributeListStandardized = [];
+					const trimmedAttribute =  this._text.trim().toLowerCase();
+
+					attributeListStandardized = this.attributeList.map((item) => {
+						return item.toLowerCase();
+					});
+					const matchedIndex = this.assignableAttributes.map((item) => {
+						return item.toLowerCase();
+					}).findIndex((x) => {
+						return x === trimmedAttribute;
+					});
+
+					if (trimmedAttribute.length > 0 && !attributeListStandardized.includes(trimmedAttribute)) {
+						if (matchedIndex >= 0) {
+							this.addAttribute(this.assignableAttributes[matchedIndex]);
+						} else {
+							this.addAttribute(this._text.trim());
+						}
 					}
```

Comment:
Let me introduce you to my friend the [Guard Clause](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html)

We can reduce the nesting of if statements here and get clearer code, as well as avoid the work of unnecessarily calculating the matchedIndex, by inverting your if statement and placing it above that work. 

```suggestion
					if (trimmedAttribute.length === 0 || attributeListStandardized.includes(trimmedAttribute)) {
						return;
					}

					const matchedIndex = this.assignableAttributes.map((item) => {
						return item.toLowerCase();
					}).findIndex((x) => {
						return x === trimmedAttribute;
					});

					if (matchedIndex >= 0) {
						this.addAttribute(this.assignableAttributes[matchedIndex]);
					} else {
						this.addAttribute(this._text.trim());
					}
```

Reply:
Aaah I rarely ever think of guard clauses - thank you!